src/modules/workspace/services/ai-assistant.service.ts

```
@@ -564,94 +386,103 @@ export class AiAssistantService {
-          TOPIC.AI_RESPONSE_GENERATED_TOPIC,
-          {
-            value: JSON.stringify(kafkaMessage),
-          },
+      if (!text) {
+        throw new BadRequestException(
+          "Invalid input: 'text' field is required.",
         );
-
-        const latestAssistantMessage = messagesResponse.data
-          .filter((message) => message.role === "assistant")
-          .sort((a, b) => new Date(b.created_at).getTime() - new Date(a.created_at).getTime())[0];
-
-        return latestAssistantMessage.content
-          .filter((item) => item.type === "text")
-          .map((item) => item.text.value)
-          .join(" ");
-      } else {
-        console.error(`Run status is ${runStatus}, unable to fetch messages.`);
-        throw new InternalServerErrorException("Assistant could not complete the request.");
       }
+      
+      const curl = text.curl;
+      const error = text.error;
+      
+      const prompt =
+      "You are provided with two things. One is cURL and second is the error message. You need to provide the solution for the error message.";
+      
+      const userMessage = `This is the cURL: ${curl} and this is the Error: ${error}`;
+      
+      const messages: { role: "system" | "user"; content: string }[] = [
+        { role: "system", content: prompt },
+        { role: "user", content: userMessage },
+      ];
+      
+      const response = await this.assistantsClient.chat.completions.create({
+        model: "sparrow",
+        messages: messages,
+        temperature: 0.7,
+        max_tokens: 1024,
+      });
+      
+      const result =
+      response.choices[response.choices.length - 1].message.content.trim();
+      return result;
     } catch (error) {
-      console.error("Error in AI Assistant:", error);
-      return "An error occurred while processing your request. Please try again.";
+      console.error("Error processing specificError:", error);
+      throw new InternalServerErrorException(
```

Why we are returning `InternalServerErrorException` from here? 
So we use this only when server in down not in case of error. For error we should use BadRequestException.

---

Never use camelCase in normal text : change specificError to specific error

---

We are throwing a BadRequestException when text is not present, we should propagate the same error from here. If there is any other error then we should throw bade request with that specific error so that we have a context.